### PR TITLE
Refactor hints and tests

### PR DESCRIPTION
**Pull request by chshersh**

### CODE REVIEW

File: test/Test/Suggestion.hs [hunk 1]
```diff
@@ -1,38 +1,24 @@
+-- TODO: rename to
+-- Test.Hint
+--   Test.Hint.HLint
+--   Test.Hint.TrailingSpaces
```
Comment (vrom911) [inline test/Test/Suggestion.hs]: Yes!

File: src/Hintman.hs [hunk 2]
```diff
@@ -44,8 +43,6 @@ runOn cli = do
     let siteString = "https://localhost:" <> show port
     putTextLn ("Starting hintman site at: " <> siteString)
 
-    envConfig <- loadFileConfig "hintman-config.toml"
```
Comment (vrom911) [inline src/Hintman.hs]: Could you also delete `hintman-config.toml` itself?

File: test/Spec.hs [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
 main = do
     hSetEncoding stdout utf8
     hSetEncoding stderr utf8
-    hspec $ do
-        configSpec
-        suggestionSpec
-        hlintSpec
-    ifM suggestionTests exitSuccess exitFailure
+
+    hspec $ hlintSpec
```
Comment (vrom911) [inline test/Spec.hs]:
```suggestion
    hspec hlintSpec
```

Comment (chshersh) [reply test/Spec.hs]: Hintman has new pretty logo already? :smirk:

File: src/Hintman.hs (hunk 2 above)
Comment (chshersh) [reply src/Hintman.hs]: Turns out there's no such file :man_shrugging:

Comment (vrom911) [reply src/Hintman.hs]: :thinking:

**Pull request merged**